packages/flutter/test/widgets/widget_state_property_test.dart

```diff
@@ -145,6 +145,86 @@ void main() {
     expect(borderSide.color, isSameColorAs(const Color(0xff0000ff)));
     expect(borderSide.width, 12.0);
   });
+
+  test('.fromMap() constructors perform accurate equality checks', () {
+    const Color white = Color(0xFFFFFFFF);
+    const Color black = Color(0xFF000000);
+    final WidgetStateColor color1 = WidgetStateColor.fromMap(
+      <WidgetStatesConstraint, Color>{
+        WidgetState.focused | WidgetState.hovered: white,
+        WidgetState.any: black,
+      },
+    );
+    final WidgetStateColor color2 = WidgetStateColor.fromMap(
+      <WidgetStatesConstraint, Color>{
+        WidgetState.focused | WidgetState.hovered: white,
+        WidgetState.any: black,
+      },
+    );
+    expect(color1 == color2, isTrue);
```

Comment:
For the purpose of watching for regressions, should check if a third, different WidgetStateColor map is unequal.

Same with all of these examples.

Reply:
ooh, yeah that'd be good.